### ISSUE TITLE
[Recommendation] Use ACRA to send crash reports without user interaction

### ISSUE THREAD

**Issue by EmmanuelMess** [TeamAmaze/AmazeFileManager#659]
[ACRA](https://github.com/ACRA/acra) with [Cloudant](https://cloudant.com/) is a nice solution, and free.

**Comment by vishal0071** [comment]
We don't have man power to manage that many crash reports. Plus, we can already get critical ones on Google Play Developer Console.

**Comment by IzzySoft** [comment]
@VishalNehra does it still hold true that ACRA is used opt-in? They changed the defaults a while ago, so I had to assign the `Tracking` anti-feature to the library. If you switched to the "prompt option" (i.e. the user is asked whether the report shall be sent), I can white-list it for Amaze. Otherwise I have to add the AF. Thanks in advance for letting me know!

**Comment by IzzySoft** [comment]
Looking [at the code](https://github.com/TeamAmaze/AmazeFileManager/blob/6eec21ae927e8826a9aff67075d36dae7dc85a1a/app/src/main/java/com/amaze/filemanager/crashreport/ErrorActivity.java#L200) it seems the user is asked where to send the report to (email, Telegram, Clipboard, Github). If that's the only place handling reports, all should be fine and I can simply whitelist ACRA for Amaze. As I'm not an Android/Java dev I can however not be sure if I might not have missed something (especially as there are multiple `public static void reportError()` methods) – so I just hope for your confirmation :wink:

**Comment by VishalNehra** [comment]
Yes, that's the only place. User has to explicitly send us reports using either of these options and we have a backend which creates GitHub issues based on reports sent by user to either email or telegram. 
Thanks @IzzySoft for looking into it.
I'll keep this issue open to remove our ACRA usage altogether.

**Comment by VishnuSanal** [comment]
@VishalNehra .

> to remove our ACRA usage altogether.

Why?

**Comment by IzzySoft** [comment]
Thanks @VishalNehra – whitelisted ACRA then for Amaze. Opt-in is totally fine, and agree with @VishnuSanal that complete removal is not necessary: the users have the choice of where to report to, and even whether to report at all (they could simply close the dialog and nothing would be sent then), right? All fine then I'd say.

**Comment by VishalNehra** [comment]
I didn't mean to say I'll remove the reporting screen. Users will still be able to report via any of the platforms, it's just that we're not using any of the features of ACRA apart from providing a hook that initiates a activity in case of any crash. That can easily be implemented without the use of a library.

**Comment by EmmanuelMess** [comment]
What would this issue be now?
